### PR TITLE
TNO-2325 Content List View Pending Transcripts

### PR DESCRIPTION
**Pull request by Fosol**
Refactored how we filter on pending transcriptions.  A first request for work orders is made, and then a subsequent request for content that exists in the work orders.

## Summary

- Updated Editor app
- Updated Content List View
- Fixed bugs with Work Order requests
- Changed Elasticsearch requests
- Published `tno-core:0.1.32`

## Content List View

image

### CODE REVIEW

File: app/editor/src/features/content/list-view/ContentListView.tsx [hunk 1]
```diff
@@ -198,53 +201,59 @@ const ContentListView: React.FC = () => {
...
-            status: WorkOrderStatusName.InProgress, // kgm: bring back only what we are interested in
+          let workOrders: IPaged<IWorkOrderModel> = {
+            page: 0,
+            quantity: 0,
+            total: 0,
+            items: [],
           };
+          if (filter.pendingTranscript) {
+            // Make a request for transcript work orders.
+            const startDate = moment().add(-3, 'day').startOf('day');
```
Comment (Fosol) [inline app/editor/src/features/content/list-view/ContentListView.tsx:212]: Fixed date range issues.  Must be beginning of day and end of day.

File: app/editor/src/features/content/list-view/ContentListView.tsx [hunk 2]
```diff
@@ -198,53 +201,59 @@ const ContentListView: React.FC = () => {
...
+            const endDate = moment().endOf('day');
+            const woFilter: IWorkOrderFilter = {
+              page: 1,
+              quantity: 1000,
```
Comment (Fosol) [inline app/editor/src/features/content/list-view/ContentListView.tsx:216]: Increased size to get all possible work orders.

File: app/editor/src/features/content/list-view/ContentListView.tsx [hunk 3]
```diff
@@ -198,53 +201,59 @@ const ContentListView: React.FC = () => {
...
+              createdStartOn: startDate.toISOString(),
+              createdEndOn: endDate.toISOString(),
+              workType: WorkOrderTypeName.Transcription,
```
Comment (Fosol) [inline app/editor/src/features/content/list-view/ContentListView.tsx:219]: We only want transcripts

File: app/editor/src/features/content/list-view/ContentListView.tsx [hunk 4]
```diff
@@ -198,53 +201,59 @@ const ContentListView: React.FC = () => {
...
+              status: WorkOrderStatusName.InProgress,
+            };
 
-          const [results, workOrders] = await Promise.all([
-            findContentWithElasticsearch(toFilter(filter), true),
-            findWorkOrders(woFilter),
-          ]);
+            const response = await findWorkOrders(woFilter);
+            workOrders = response.data;
+            filter.contentIds = workOrders.items
```
Comment (Fosol) [inline app/editor/src/features/content/list-view/ContentListView.tsx:225]: Apply the results to the filter for elasticsearch

File: app/editor/src/features/content/list-view/ContentListView.tsx [hunk 5]
```diff
@@ -198,53 +201,59 @@ const ContentListView: React.FC = () => {
...
+              .filter((wo) => !!wo.contentId)
+              .map((wo) => wo.contentId!);
+          }
 
-          const items = results.hits?.hits?.map((h) =>
+          const doSearch =
+            !filter.pendingTranscript || (filter.pendingTranscript && workOrders.items.length);
+          const searchResults: KnnSearchResponse<IContentModel> = doSearch
```
Comment (Fosol) [inline app/editor/src/features/content/list-view/ContentListView.tsx:232]: Don't bother making a request if there are no work orders

File: app/editor/src/features/content/list-view/ContentListView.tsx [hunk 6]
```diff
@@ -198,53 +201,59 @@ const ContentListView: React.FC = () => {
...
+          let items = searchResults.hits?.hits?.map((h) =>
             castContentToSearchResult(h._source as IContentModel),
           );
 
-          if (filter.pendingTranscript && items) {
-            let itemsWithStatus: IContentSearchResult[] | undefined = items.map((m) => {
-              let v = { ...m };
-              v.transcriptStatus = workOrders.data.items.find((w) => w.contentId === m.id)?.status;
-              return v;
+          if (filter.pendingTranscript) {
```
Comment (Fosol) [inline app/editor/src/features/content/list-view/ContentListView.tsx:241]: Simplify!

File: app/editor/src/features/content/list-view/components/tool-bar/filter/InputOption.tsx [hunk 7]
```diff
@@ -49,7 +49,7 @@ export const InputOption: React.FC<any> = ({
       getStyles={getStyles}
       innerProps={props}
     >
-      <input type="checkbox" id={children} checked={isSelected} />
+      <input type="checkbox" id={children} defaultChecked={isSelected} />
```
Comment (Fosol) [inline app/editor/src/features/content/list-view/components/tool-bar/filter/InputOption.tsx:52]: Fixed bug

File: app/editor/src/features/content/list-view/components/tool-bar/filter/ShowOnlySection.tsx [hunk 8]
```diff
@@ -200,7 +200,7 @@ export const ShowOnlySection: React.FC<IShowOnlySectionProps> = () => {
                 name="pendingTranscript"
                 label="Pending Transcript"
                 tooltip="Pending Transcript"
-                checked={filter.pendingTranscript}
+                checked={filter.pendingTranscript ?? false}
```
Comment (Fosol) [inline app/editor/src/features/content/list-view/components/tool-bar/filter/ShowOnlySection.tsx:203]: Fixed bug.  React state hates `undefined`

File: app/editor/src/features/content/list-view/hooks/useColumns.tsx [hunk 9]
```diff
@@ -121,7 +121,10 @@ export const useColumns = ({ fetch }: IColumnProps): ITableHookColumn<IContentSe
       width: '55px',
       cell: (cell) => <Status value={cell.original.status} />,
     },
-    {
+  ];
+
+  if (filter.pendingTranscript) {
```
Comment (Fosol) [inline app/editor/src/features/content/list-view/hooks/useColumns.tsx:126]: Only show column when filtering by pending transcriptions

File: app/subscriber/src/features/my-minister/MyMinister.tsx [hunk 10]
```diff
@@ -194,7 +194,7 @@ export const MyMinister: React.FC = () => {
               label={`${m.name} (${m.contentCount})`}
               className="option"
               checked={!m.hide}
-              onClick={(e) => {
+              onChange={(e) => {
```
Comment (Fosol) [inline app/subscriber/src/features/my-minister/MyMinister.tsx:197]: Fixed bug.  `checked` and `onChange` must be paired

**Review by kylgarmor — APPROVED** [review]

File: libs/npm/core/src/hooks/api/interfaces/IWorkOrderFilter.ts [hunk 11]
```diff
@@ -3,7 +3,7 @@ import { WorkOrderStatusName, WorkOrderTypeName } from '../constants';
 export interface IWorkOrderFilter {
   keywords?: string;
   isApproved?: boolean;
-  status?: WorkOrderStatusName;
+  status?: WorkOrderStatusName[];
```
Comment (Fosol) [inline libs/npm/core/src/hooks/api/interfaces/IWorkOrderFilter.ts:6]: Switched to array to support filtering on more than one value.

**Pull request merged**